src/features/applications/hooks/slug.ts

```diff
@@ -60,16 +64,26 @@ hooks.addPureHook('PATCH', 'resin', 'application', {
 				},
 			} as const);
 
+			// Attempt to get the slug for each app
+			const appSlugs = new Map<number, string>();
+			try {
+				apps.forEach((app) => {
+					appSlugs.set(
+						app.id,
+						getApplicationSlug(app.organization[0].handle, app.app_name),
+					);
+				});
+			} catch (e) {
+				throw new errors.NotFoundError(e.message);
```

Comment:
This might be a legitimate 500 actually, since we are passing in the handle & the app_name that we retrieved from the DB, and there shouldn't be a way for them to be nullable.
So I think we can revert this part.

Reply:
Won't revert it, but will make it an explicit `errors.InternalServerError(e.message);`